[2024-02-26T08:24:35.072+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: exchange_trade_data.write_to_motherduck_from_data_frame manual__2024-02-26T08:24:24.690924+00:00 [queued]>
[2024-02-26T08:24:35.091+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: exchange_trade_data.write_to_motherduck_from_data_frame manual__2024-02-26T08:24:24.690924+00:00 [queued]>
[2024-02-26T08:24:35.092+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2024-02-26T08:24:35.323+0000] {taskinstance.py:2191} INFO - Executing <Task(_PythonDecoratedOperator): write_to_motherduck_from_data_frame> on 2024-02-26 08:24:24.690924+00:00
[2024-02-26T08:24:35.329+0000] {standard_task_runner.py:60} INFO - Started process 6349 to run task
[2024-02-26T08:24:35.333+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'exchange_trade_data', 'write_to_motherduck_from_data_frame', 'manual__2024-02-26T08:24:24.690924+00:00', '--job-id', '83', '--raw', '--subdir', 'DAGS_FOLDER/exchange.py', '--cfg-path', '/tmp/tmppub3gn67']
[2024-02-26T08:24:35.336+0000] {standard_task_runner.py:88} INFO - Job 83: Subtask write_to_motherduck_from_data_frame
[2024-02-26T08:24:35.395+0000] {task_command.py:423} INFO - Running <TaskInstance: exchange_trade_data.write_to_motherduck_from_data_frame manual__2024-02-26T08:24:24.690924+00:00 [running]> on host f5283f9a0a92
[2024-02-26T08:24:35.574+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='exchange_trade_data' AIRFLOW_CTX_TASK_ID='write_to_motherduck_from_data_frame' AIRFLOW_CTX_EXECUTION_DATE='2024-02-26T08:24:24.690924+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-26T08:24:24.690924+00:00'
[2024-02-26T08:24:35.577+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/exchange.py", line 95, in write_to_motherduck_from_data_frame
    curr= con.cursor()
AttributeError: 'PlainXComArg' object has no attribute 'cursor'
[2024-02-26T08:24:35.587+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=exchange_trade_data, task_id=write_to_motherduck_from_data_frame, execution_date=20240226T082424, start_date=20240226T082435, end_date=20240226T082435
[2024-02-26T08:24:35.602+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 83 for task write_to_motherduck_from_data_frame ('PlainXComArg' object has no attribute 'cursor'; 6349)
[2024-02-26T08:24:35.627+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-02-26T08:24:35.649+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
